<a href="https://colab.research.google.com/github/Epilef86/DNC/blob/main/Formulario_Clientes.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
nomes = []
agenda = {}

while True:
    ordem = input('Deseja adicionar um novo contato(add) ou parar a execução(stop)').lower()

    if not ordem.isalpha():
        print("Digite apenas letras!")
        
    elif ordem.startswith('a'):
        contato = {}


        nome = input('Digite o nome do contato: ')
        if not nome[0].isupper():
            nome = nome[0].upper() + nome[1:]

        nomes.append(nome)

        tel = input('Digite o telefone do contato: ')
        contato['Telefone'] = tel

        end = input('Digite o endereço do contato: ')
        contato['Endereço'] = end

        email = input('Digite o Email do contato: ')
        contato['Email'] = email

        agenda[nome] = contato
        print(agenda)
        
    elif ordem.startswith('s'):
        break

print('A G E N D A\n')
nomes.sort()
for nome in nomes:
    print('Nome: ', nome)
    print('Telefone: ', agenda[nome]['Telefone'])
    print('Endereço: ', agenda[nome]['Endereço'])
    print('Email: ', agenda[nome]['Email'])
    print()

Deseja adicionar um novo contato(add) ou parar a execução(stop)add
Digite o nome do contato: Felipe
Digite o telefone do contato: 81995294484
Digite o endereço do contato: Rua Desembargador Virgílio de Sá Pereira
Digite o Email do contato: amaral.felipelima@gmail.com
{'Felipe': {'Telefone': '81995294484', 'Endereço': 'Rua Desembargador Virgílio de Sá Pereira', 'Email': 'amaral.felipelima@gmail.com'}}
Deseja adicionar um novo contato(add) ou parar a execução(stop)add
Digite o nome do contato: Mizael
Digite o telefone do contato: 81999248583
Digite o endereço do contato: Rua Veneza
Digite o Email do contato: mizael.correia@gmail.com
{'Felipe': {'Telefone': '81995294484', 'Endereço': 'Rua Desembargador Virgílio de Sá Pereira', 'Email': 'amaral.felipelima@gmail.com'}, 'Mizael': {'Telefone': '81999248583', 'Endereço': 'Rua Veneza', 'Email': 'mizael.correia@gmail.com'}}
Deseja adicionar um novo contato(add) ou parar a execução(stop)stop
A G E N D A

Nome:  Felipe
Telefone:  81995294484
Ender

nomes = []
agenda = {}

while True:
    ordem = input('Deseja adicionar um novo contato(add) ou parar a execução(stop)').lower()
    ordem = add

    if not ordem.isalpha():
    if not True
    if False:

'''Sempre que o resultado dessa comparação resulta em TRUE, ele entra na condição 
e executa o código,caso seja FALSE, ele não executa o bloco, 
podendo cair no bloco de um ELSE'''

       print("Digite apenas letras!")
        
    elif ordem.startswith('a'):
    elif True
        contato = {}

''' .startswith --> Verifique se a string começa com 'a''''

	nome = input('Digite o nome do contato: ')
	nome = Felipe

        if not nome[0].isupper():
	if not 'F' é maiúscula?
	if not True
	if False:

'''nome[0].isupper() --> se a 1ª letra não for maiúscula, executa. 

            nome = nome[0].upper() + nome[1:]

        nomes.append(nome)
	nomes = ['Felipe']

        tel = input('Digite o telefone do contato: ')
	tel = 81995294484

        contato['Telefone'] = tel
	contato = {'Telefone':81995294484}

        end = input('Digite o endereço do contato: ')
	end = Rua Desembardador Virgílio de Sá Pereira

        contato['Endereço'] = end
	contato = {'Telefone':81995294484, 'Endereço':'Rua Desembardador Virgílio de Sá Pereira'}

        email = input('Digite o Email do contato: ')
	email = amaral.felipelima@gmail.com

        contato['Email'] = email

	contato = {
	'Telefone':81995294484, 
	'Endereço':'Rua Desembardador Virgílio de Sá Pereira',
	'Email':'amaral.felipelima@gmail.com 
	}

        agenda[nome] = contato
	agenda['Felipe'] = {
	'Telefone':81995294484, 
	'Endereço':'Rua Desembardador Virgílio de Sá Pereira',
	'Email':'amaral.felipelima@gmail.com 
	}

------------------------------------------------------------------------------------------------------
linha 10:

while True:
    ordem = input('Deseja adicionar um novo contato(add) ou parar a execução(stop)').lower()
    ordem = add

    if not ordem.isalpha():
    if not True
    if False:

       print("Digite apenas letras!")
        
    elif ordem.startswith('a'):
    elif True
        contato = {}

''' .startswith --> Verifique se a string começa com 'a''''

	nome = input('Digite o nome do contato: ')
	nome = Felipe

        if not nome[0].isupper():
	if not 'F' é maiúscula?
	if not True
	if False:

'''nome[0].isupper() --> se a 1ª letra não for maiúscula, executa. 

            nome = nome[0].upper() + nome[1:]

        nomes.append(nome)
	nomes = ['Felipe', 'Mizael']

Observe que contato = {}, ou seja, zerou as informações adicionadas anteriormente

        tel = input('Digite o telefone do contato: ')
	tel = 81999248583

        contato['Telefone'] = tel
	contato = {'Telefone':81999248583}

        end = input('Digite o endereço do contato: ')
	end = Rua Veneza
        contato['Endereço'] = end
	contato = {'Telefone':81999248583, 'Endereço':'Rua Veneza'}

        email = input('Digite o Email do contato: ')
	email = mizael.correia@gmail.com

        contato['Email'] = email

	contato = {
	'Telefone':81999248583, 
	'Endereço':'Rua Veneza',
	'Email':'mizael.correia@gmail.com' 
	}

        agenda[nome] = contato

	agenda['Felipe', 'Mizael'] = {[
	'Telefone':81995294484, 
	'Endereço':'Rua Desembardador Virgílio de Sá Pereira',
	'Email':'amaral.felipelima@gmail.com'], 
	['Telefone':81999248583, 
	'Endereço':'Rua Veneza',
	'Email':'mizael.correia@gmail.com'] 
	}

----------------------------------------------------------------------------------------------------
'''O método append não existe na classe dict(). 
O que você pode fazer é utilizar uma chave dentro de colchetes 
no primeiro dicionário e depois atribuir o outro dicionário a essa chave.

dicionario["<chave>"] = outro_dicionario '''

-----------------------------------------------------------------------------------------------------
linha 10:

while True:
    ordem = input('Deseja adicionar um novo contato(add) ou parar a execução(stop)').lower()
    ordem = stop

-------------------------------------------------------------------------------------------------------
linha 37:

     elif ordem.startswith('s'):
     elif ordem começa com a letra 's'? True
        break
-------------------------------------------------------------------------------------------------------
linha 40:

print('A G E N D A\n')
nomes.sort()                -----> colocar na ordem alfabética

for nome in nomes:
for 'Felipe' in ['Felipe', 'Mizael']:

    print('Nome: ', nome)
    'Nome': 'Felipe'
    print('Telefone: ', agenda[nome]['Telefone'])
    agenda['Felipe'][81995294484]
    Telefone: 
    
    print('Endereço: ', agenda[nome]['Endereço'])
    agenda['Felipe']['Rua Desembardador Virgílio de Sá Pereira']
    Endereço: 'Rua Desembardador Virgílio de Sá Pereira'

    print('Email: ', agenda[nome]['Email'])
    agenda['Felipe']['amaral.felipelima@gmail.com']
    Email: 'amaral.felipelima@gmail.com'
    print()

--------------------------------------------------------------------------------------------------------
Idem para Mizael:

for 'Mizael' in ['Felipe', 'Mizael']:

    print('Nome: ', nome)
    print('Telefone: ', agenda[nome]['Telefone'])
    print('Endereço: ', agenda[nome]['Endereço'])
    print('Email: ', agenda[nome]['Email'])
    print()